Client/Frontend/Browser/URLBarView.swift

```diff
@@ -455,6 +457,15 @@ class URLBarView: UIView {
         }
     }
 
+    func updateTabsButtonHighlight() {
+        tabsButton.textColor = UIColor.whiteColor().isEqual(tabsButton.textColor) ? TabsButtonUX.TitleColor : UIColor.whiteColor()
+
+        previousTabsButtonColor = UIColor.whiteColor().isEqual(tabsButton.textColor) ? tabsButton.titleBackgroundColor! : previousTabsButtonColor
+        tabsButton.titleBackgroundColor = UIColor.whiteColor().isEqual(tabsButton.textColor) ? UIColor.grayColor() : previousTabsButtonColor
+
+        tabsButton.borderColor = UIColor.whiteColor().isEqual(tabsButton.textColor) ? UIColor.whiteColor() : UIColor.blackColor()
+    }
```

Comment:
This code is pretty confusing. I'd redesign it:
1. In `TabsButton`, create a `selected` boolean.
2. Add a `didSet` observer for the boolean that updates the text, title, and background colors based on the boolean state.
3. Update the `selected` boolean based on the control events.

That way, the logic is contained in `TabsButton` and we have real state instead of this confusing `UIColor.whiteColor().isEqual(tabsButton.textColor)` condition.